pkg/registry/core/service/storage/rest.go

```diff
@@ -846,9 +859,77 @@ func (rs *REST) releaseServiceClusterIPs(service *api.Service) (released map[api
 	return rs.releaseClusterIPs(toRelease)
 }
 
+// tests if two prefered dual-stack service have matching ClusterIPFields
+// assumption: old service is a valid, default service (e.g., loaded from store)
+func isMatchingPreferDualStackClusterIPFields(oldService, service *api.Service) bool {
+	if oldService == nil {
+		return false
+	}
+
+	if service.Spec.IPFamilyPolicy == nil {
+		return false
+	}
+
+	// both must be type == ClusterIP
+	if service.Spec.Type != api.ServiceTypeClusterIP {
+		return false
+	}
+
+	if oldService.Spec.Type != api.ServiceTypeClusterIP {
+		return false
+	}
```

Comment:
NodePort and LoadBalancer types does have ClusterIPs too, should this be?

"if Type in [ClusterIP, NodePort, Loadbalancer] && and sevice.Spec.ClusterIP is an IP address (avoid headless services with clusterip=None"

Reply:
I think that if you change the service type on purpose it makes sense to add/remove the ClusterIP, but not if you keep the same type (i.e, I add a label to a loadbalancer, nodeport or clusterIP service)